x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -404,18 +407,11 @@ private void authorizeAction(
             }, clusterAuthzListener::onFailure));
         } else if (isIndexAction(action)) {
             final Metadata metadata = clusterService.state().metadata();
-            final AsyncSupplier<Set<String>> authorizedIndicesSupplier = new CachingAsyncSupplier<>(authzIndicesListener -> {
-                Consumer<Collection<String>> timeChecker = authzIndicesTimerFactory.newTimer(requestInfo);
-                authzEngine.loadAuthorizedIndices(
-                    requestInfo,
-                    authzInfo,
-                    metadata.getIndicesLookup(),
-                    authzIndicesListener.map(authzIndices -> {
-                        timeChecker.accept(authzIndices);
-                        return authzIndices;
-                    })
-                );
-            });
+            final AsyncSupplier<Set<String>> authorizedIndicesSupplier = new CachingAsyncSupplier<>(
```

Comment:
Do you think we still need the `CachingAsyncSupplier` wrapper magic?
I would ditch it, it's hard to follow and useless now.

Reply:
I agree it is redundant now and should be removed.